crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -138,28 +185,32 @@ impl Object {
 
     /// The Base64 encoded bcs serialization of the object's content.
     async fn bcs(&self) -> Result<Option<Base64>> {
-        if let Some(stored) = &self.stored {
-            Ok(Some(Base64::from(&stored.serialized_object)))
-        } else {
-            let bytes = bcs::to_bytes(&self.native)
-                .map_err(|e| {
-                    Error::Internal(format!(
-                        "Failed to serialize object at {}: {e}",
-                        self.address,
-                    ))
-                })
-                .extend()?;
-
-            Ok(Some(Base64::from(&bytes)))
-        }
+        let Some(native) = self.state.native() else {
```

Comment:
Folding the `stored` field into `state` (or `kind` after the suggested change above) would allow us to keep this optimisation.